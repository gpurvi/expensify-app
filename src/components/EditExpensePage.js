import React from 'react'
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense, removeExpense} from "../actions/expenses";

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    onClick = () => {
        this.props.removeExpense(this.props.expense);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button
                    onClick={this.onClick}
                >Remove
                </button>
            </div>
        );
    }
}

const mapStateTopProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch) => ({
    editExpense: (id, updates) => dispatch(editExpense(id, updates)),
    removeExpense: (expense) => dispatch(removeExpense(expense))
});

export default connect(mapStateTopProps, mapDispatchToProps)(EditExpensePage);