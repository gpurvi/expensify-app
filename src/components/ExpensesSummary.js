import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectTotalExpenses from '../selectors/expenses-total';

export const ExpensesSummary = (props) => {
    const length = props.expenses.length;
    const formatTotal = (selectTotalExpenses(props.expenses)/100).toFixed(2);
    const includeS = length <= 1 ? '' : 's';
    return (<h1>Viewing {length} expense{includeS} totalling ${formatTotal}</h1>);
};

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);

