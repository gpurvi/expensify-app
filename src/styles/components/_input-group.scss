.input-group {
  display: flex;
  margin-bottom: $m-size;
  flex-direction: column;
  @media (min-width: $desktop-breakpiont) {
    flex-direction: row;
    margin-bottom: $l-size;
  }
}

.input-group__item {
  margin-right: $s-size;
  margin-bottom: $s-size;
  @media (min-width: $desktop-breakpiont) {
    margin: 0 $s-size 0 0;
  }
}