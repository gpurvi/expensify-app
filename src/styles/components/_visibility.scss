.show-for-mobile {
  @media (min-width: $desktop-breakpiont) {
    display: none;
  }
}

.show-for-desktop {
  @media (max-width: $desktop-breakpiont - .01rem) {
    display: none;
  }
}