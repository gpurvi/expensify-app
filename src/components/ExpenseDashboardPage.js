import React from 'react';
import ExpensList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from "./ExpensesSummary";

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary/>
        <ExpenseListFilters/>
        <ExpensList/>
    </div>
);

export default ExpenseDashboardPage;
