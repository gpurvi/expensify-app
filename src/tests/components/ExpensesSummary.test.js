import React from 'react';
import {ExpensesSummary} from "../../components/ExpensesSummary";
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary correctly with no expenses', () => {
    const emptyExpenses = [];
    const wrapper = shallow(<ExpensesSummary expenses={emptyExpenses}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with 1 expense', () => {
    const oneExpense = [expenses[0]];
    const wrapper = shallow(<ExpensesSummary expenses={oneExpense}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with 3 expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});