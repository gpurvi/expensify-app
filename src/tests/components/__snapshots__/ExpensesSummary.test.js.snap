// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with 1 expense 1`] = `
<h1>
  Viewing 
  1
   expense
   totalling $
  1.95
</h1>
`;

exports[`should render ExpensesSummary correctly with 3 expenses 1`] = `
<h1>
  Viewing 
  3
   expense
  s
   totalling $
  10996.95
</h1>
`;

exports[`should render ExpensesSummary correctly with no expenses 1`] = `
<h1>
  Viewing 
  0
   expense
   totalling $
  0.00
</h1>
`;
